components/sup/src/manager/service/dir.rs

```diff
@@ -136,7 +147,11 @@ impl<'a> SvcDir<'a> {
         }
     }
 
-    fn create_dir_all<P: AsRef<Path>>(path: P) -> Result<()> {
+    fn create_dir_all<P: AsRef<Path>>(path: P) -> Result<CreateDirectoryStatus> {
```

Comment:
`create_dir_all` has quite a few callers, but if only this one case cares about this newly added return type, I'd rather leave `create_dir_all` alone and add the `exists()` check to the caller side. It should be less code and complexity and eliminates the need for the new `CreateDirectoryStatus` type (though I do really appreciate the use of the more readable `enum` over a bool).

I think the change could be as simple as
```rust
    fn create_svc_owned_dir<P>(&self, path: P) -> Result<()>
    where
        P: AsRef<Path>,
    {
        if path.as_ref().exists() {
            Ok(())
        } else {
            // Original body of create_svc_owned_dir
        }
    }
```

Reply:
I agree with @baumanj here, but particularly on the use "custom boolean enum" pattern; that's one of my favorite things, and I'd like to see a lot more of it!
![tenor-82140374](https://user-images.githubusercontent.com/207178/46675528-9f2add00-cbac-11e8-8d7f-b482241ac1a6.gif)